fast64_internal/oot/oot_actor.py

```diff
@@ -1,47 +1,410 @@
+		for obj in bpy.context.scene.objects:
+			if obj.fast64.oot.version != self.cur_version:
+				OOTActorProperties.upgrade_object(obj)
+			obj.fast64.oot.version = self.cur_version
+		return {'FINISHED'}
+
+class OOT_SearchChestContentEnumOperator(bpy.types.Operator):
+	bl_idname = "object.oot_search_chest_content_enum_operator"
+	bl_label = "Select Chest Content"
+	bl_property = "itemChest"
+	bl_options = {'REGISTER', 'UNDO'}
+
+	itemChest : bpy.props.EnumProperty(items = ootChestContent, default = '0x48')
 	objName : bpy.props.StringProperty()
 
 	def execute(self, context):
-		obj = bpy.data.objects[self.objName]
-		if self.actorUser == "Transition Actor":
-			obj.ootTransitionActorProperty.actor.actorID = self.actorID
-		elif self.actorUser == "Actor":
-			obj.ootActorProperty.actorID = self.actorID
-		elif self.actorUser == "Entrance":
-			obj.ootEntranceProperty.actor.actorID = self.actorID
-		else:
-			raise PluginError("Invalid actor user for search: " + str(self.actorUser))
+		bpy.data.objects[self.objName].fast64.oot.actor.itemChest = self.itemChest
+		bpy.context.region.tag_redraw()
+		self.report({'INFO'}, "Selected: " + self.itemChest)
+		return {'FINISHED'}
 
+	def invoke(self, context, event):
+		context.window_manager.invoke_search_popup(self)
+		return {'RUNNING_MODAL'}
+
+class OOT_SearchNaviMsgIDEnumOperator(bpy.types.Operator):
+	bl_idname = "object.oot_search_navi_msg_id_enum_operator"
+	bl_label = "Select Message ID"
+	bl_property = "naviMsgID"
+	bl_options = {'REGISTER', 'UNDO'}
+
+	naviMsgID : bpy.props.EnumProperty(items = ootNaviMsgID, default = "0x00")
+	objName : bpy.props.StringProperty()
+
+	def execute(self, context):
+		bpy.data.objects[self.objName].fast64.oot.actor.naviMsgID = self.naviMsgID
 		bpy.context.region.tag_redraw()
-		self.report({'INFO'}, "Selected: " + self.actorID)
+		self.report({'INFO'}, "Selected: " + self.naviMsgID)
 		return {'FINISHED'}
 
 	def invoke(self, context, event):
 		context.window_manager.invoke_search_popup(self)
 		return {'RUNNING_MODAL'}
 
+class OOTActorProperties(bpy.types.PropertyGroup):
+	@staticmethod
+	def upgrade_object(obj):
+		if obj.data is None:
+			if obj.ootEmptyType == 'Scene':
+				print(f"Processing Scene '{obj.name}'...")
+				for roomObj in obj.children:
+					if roomObj.data is None:
+						if roomObj.ootEmptyType == 'Room':
+							# Actor is parented to a room
+							for obj in roomObj.children:
+								upgradeActorInit(obj)
+						elif (roomObj.ootEmptyType == 'Actor' or \
+							roomObj.ootEmptyType == 'Transition Actor' or roomObj.ootEmptyType == 'Entrance'):
+							# Actor is parented to a scene
+							upgradeActorInit(roomObj)
+			elif (obj.ootEmptyType == 'Actor' or obj.ootEmptyType == 'Transition Actor' or obj.ootEmptyType == 'Entrance'):
+				upgradeActorInit(obj)
+
+def getValues(self, actorID, actorField, target, paramField):
+	'''Updates the actor parameter field when the user change the options'''
+	if self.isActorSynced:
```

Comment:
instead of `isActorSynced` shouldn't this be a `version == cur_version` check?

having an "is up-to-date" boolean is a problem because obviously it has two values so it's not friendly for reusing in a future update. and it gets obsoleted immediately